status-im/status-react
# Replace `GetWalletToken` endpoint with `FetchOrGetCachedWalletBalances`

fixes #20473

### Summary
`GetWalletToken` endpoint in `status-go` was replaced by `FetchOrGetCachedWalletBalances`
Visible impact - when user drags down wallet screen and triggers refresh, it happens much quicker - fraction of second instead of few seconds. 

### Review notes
Deprecated `GetWalletToken` function was packing a big chunk of information together - not only token balances but also their prices and market values. That resulted in data duplication - inside tokens we had the same information about prices and market values as we already have in re-frame db under `[:wallet :tokens :prices-per-token]` and `[:wallet :tokens :market-values-per-token]`. Moreover, in time that information became out of sync, because data under `[:wallet :tokens ...]` never updated after wallet initialisation while data inside token structure updated regularly.

This PR replaces the endpoint and removes the data duplication.

Token structures fetched with `FetchOrGetCachedWalletBalances` do not have information about prices and market values. But this information can be found under `[:wallet :tokens :prices-per-token]` and `[:wallet :tokens :market-values-per-token]`, so big amount of small changes is a result of necessity to pass another structures with prices and market values apart from token itself.

When `FetchOrGetCachedWalletBalances` finishes successfully we now additionally request prices and market values for those tokens that exist in wallet to have an up-to-date data after user triggered refresh.

### Testing notes
From user perspective this PR should't change anything except the speed of wallet refresh when user drags screen down. It is faster now.

#### Platforms
- Android
- iOS

#### Areas that maybe impacted

##### Functional

- wallet / transactions

### Steps to test

- Open Status
- Check all screens where prices
- Make sure prices are displayed and displayed correctly




status: ready
